### ISSUE TITLE
Playback latency

### ISSUE THREAD

**Issue by githubjanson** [jfversluis/Plugin.Maui.Audio#89]
In my mobile app, I need a short sound to play when I tap a button. And I need as good precision as possible, that is, as short latency between the tap and the sound as possible.

But the latency is approximately 150-200 ms when running the app on a (modern) iPhone, which is not good enough for my application. There are similar apps out there (non-MAUI), where this lag is significantly less, so it is not hardware related.

- I am using a really small wav (7 kB).
- I am preparing the audio player once, loading the stream when the app loads.
- The only action when tapping the button is invoking Play() or PlayAsync().
- I am running the app on device (no emulator).
- Tried running in Release mode without debugging, no (noticable) difference.

I assume that this lag is not related to the Maui.Audio plugin, but perhaps someone has a clue or workaround how to get around this issue? Do I need to switch to another platform?

Thanks.

**Comment by bijington** [comment]
It is interesting that you are seeing some latency. Out of interest how are you measuring this latency?

A couple of things that I would try:
1. Check that your sound starts right at the beginning of the file and there isn't any silence mimicking latency.
2. If you are using a Command/Clicked event swap to using the Pressed event on the Button to see if you can get a quicker response.

**Comment by githubjanson** [comment]
@bijington, thanks for your tips! Changing to the Pressed event didn't help but made me realize that the latency is in the Button control and not related to the sound playback at all. When I press the button (and hold) there is a delay before the button "reacts". For the default MAUI button this is shown by the text turning grey. And at that exact moment, my sound plays.

So what I need is a button with shorter response time.

Regarding the latency, I have not measured it but rather estimated it by comparison. I am developing a rythmic exercise app, and I guess a latency around 50 ms would be acceptable, 25 ms would be great.

**Comment by borrmann** [comment]
@githubjanson out of curiosity did you find a solution for this issue? 
If not, are you using compiled bindings or have you filed an issue in the MAUI repo?

**Comment by its-jefe** [comment]
Seems unlikely a problem with Plugin.Maui.Audio in my case. Other MAUI elements don't seem to have a latency but I also noticed a delay on button clicks. I had tried calling play from event triggers of pressed and clicked. Then from inside a custom button OnPressed. Also curious to know @githubjanson if you found a solution.

**Comment by githubjanson** [comment]
Hi @borrmann, @its-jefe,

I skipped the Button control and am using the SKCanvasView/Touch event instead. This gives me the best experience, i.e., a minimal latency beteen the tap and the sound.

**Comment by jfversluis** [comment]
Looks there is a solution. For a next time it would be super helpful if you could include a reproduction project so we can have a look together with you and easily see if this reproduces on our end as well as well as maybe fix it. For now closing this. If there is still something that needs to be investigated here, please let us know!

**Comment by Landenater02** [comment]
@githubjanson Hey, I'm trying to make an app with metronome tap functionality - do you think you could share what you had here? I know it's been a while, but I've tried using the skia canvas touch event as you said, and I'm still getting some latency in my project and having a tough time figuring it out. Thank you!

**Comment by githubjanson** [comment]
Hi @Landenater02, I ended up using the NAudio library, it's been a while and I don't have the details top of mind right now. But I chose that library because it was much more powerful wrt to features and potential (but also more complex). I did a metronome project with a Tap feature, just as a first MAUI iOS app project. And it worked fine, finally. Now, looking into the code, I see that I used the SKCanvasView for the tapping:

`<controls:SKCanvasView
    x:Name="Tap"
    HeightRequest="64"
    WidthRequest="64"
    BackgroundColor="Transparent"
    EnableTouchEvents="True"
    Touch="Tap_OnTouch"/>`
